varats/varats/tools/driver_casestudy.py

```diff
@@ -561,8 +568,32 @@ def find_result_dir_paths_of_projects() -> tp.List[Path]:
         return existing_paper_config_result_dir_paths
 
     def remove_old_result_files() -> None:
-        pass
-        # TODO: Implement the removal of old result files.
+        paper_config = get_paper_config()
+        result_dir = Path(str(vara_cfg()['result_dir']))
+        for case_study in paper_config.get_all_case_studies():
+            old_files = []
+            newer_files: tp.Dict[str, Path] = dict()
+            result_dir_cs = result_dir/case_study.project_name
+            if not result_dir_cs.exists():
+                continue
+            for opt_res_file in result_dir_cs.iterdir():
+                commit_hash = MetaReport.get_commit_hash_from_result_file(opt_res_file.name)
+                if case_study.has_revision(commit_hash):
+                    current_file = newer_files.get(commit_hash)
+                    if current_file is None:
+                        newer_files.__setitem__(commit_hash,opt_res_file)
+                    else:
+                        if (
+                                current_file.stat().st_mtime <
+                                opt_res_file.stat().st_mtime
+                        ):
+                            nnewer_files.__setitem__(commit_hash,opt_res_file)
+                            old_files.append(current_file)
+                        else:
+                            old_files.append(opt_res_file)
+            for file in old_files:
+                if os.path.exists(file):
```

Comment:
As you already have a `Path` here, prefer the pathlib API over `os`:
```suggestion
                if file.exists():
```